server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -457,19 +473,23 @@ onMessage authMode serverEnv wsConn msgRaw =
                            wsConn authMode params
       CMStart startMsg  -> liftIO $ onStart serverEnv wsConn startMsg
       CMStop stopMsg    -> liftIO $ onStop serverEnv wsConn stopMsg
+      -- The idea is cleanup will be handled by 'onClose', but...
+      --   TODO I don't think closeConn here is correct. Document what's going on here.
```

Comment:
Do you have any ideas? This is what we have: we need to close the websocket connection when we receive the `CMConnTerm` message and calling `WS.closeConn` will definitely throw an exception.

Reply:
My understanding looking at the docs was that `closeConn` is supposed to be a friendly way of initiating connection closed by server, so it seems weird to do this after getting a connection termination from client. But I don't know! I think best is to just add the info you commented and keep as TODO